osu.Game/Extensions/ContainerExtensions.cs

```diff
@@ -0,0 +1,215 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using osu.Framework.Extensions.EnumExtensions;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osuTK;
+
+namespace osu.Game.Extensions
+{
+    public static class ContainerExtensions
+    {
+        /// <summary>
+        /// Prevents this container's children from transforming when the container transforms.
+        /// </summary>
+        /// <param name="target">The target to prevent its children from transforming.</param>
+        /// <param name="parent">The parent of the target.</param>
+        /// <param name="previousParentRotation">The parent's rotation in the previous update cycle.</param>
+        /// <param name="previousParentScale">The parent's scale in the previous update cycle.</param>
+        public static void KeepChildrenUpright(this Container target, Container parent, float previousParentRotation, Vector2 previousParentScale)
+        {
+            float parentRotation = parent.Rotation;
+            Vector2 parentScale = parent.Scale;
+
+            foreach (Drawable child in target)
+            {
+                child.RotateTo(-parentRotation);
```

Comment:
Setting `.Rotation` and `.Scale` is preferred instead of calling `.RotateTo()` and `.ScaleTo()`, as the latter methods allocate transforms for no real reason if the change is instantaneous anyway.